### PR TITLE
Update ERFA to 1.7.1

### PR DESCRIPTION
**Pull request by helgee**

### CODE REVIEW

**Comment by helgee** [comment]
Odd failures 🤔

File: E/ERFA/build_tarballs.jl [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
 sources = [
-    ArchiveSource("https://github.com/liberfa/erfa/releases/download/v1.7.0/erfa-1.7.0.tar.gz", "f0787e30e848750c0cbfc14827de6fc7f69a2d5ef0fc653504e74b8967a764e0")
+    ArchiveSource("https://github.com/liberfa/erfa/releases/download/v1.7.1/erfa-1.7.1.tar.gz", "80239ae9ba4bf7f328f7f64baec7fc38e587058b3ef9cea9331c751de0b98783")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
-cd erfa-1.7.0/
+cd erfa-1.7.1/
```
Comment (giordano) [inline E/ERFA/build_tarballs.jl:16]:
```suggestion
cd erfa-*/
```

File: E/ERFA/build_tarballs.jl [hunk 2]
```diff
@@ -3,17 +3,17 @@
 using BinaryBuilder, Pkg
 
 name = "ERFA"
-version = v"1.7.0"
+version = v"1.7.1"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/liberfa/erfa/releases/download/v1.7.0/erfa-1.7.0.tar.gz", "f0787e30e848750c0cbfc14827de6fc7f69a2d5ef0fc653504e74b8967a764e0")
+    ArchiveSource("https://github.com/liberfa/erfa/releases/download/v1.7.1/erfa-1.7.1.tar.gz", "80239ae9ba4bf7f328f7f64baec7fc38e587058b3ef9cea9331c751de0b98783")
```
Comment (giordano) [inline E/ERFA/build_tarballs.jl:10]:
```suggestion
    ArchiveSource("https://github.com/liberfa/erfa/releases/download/v$(version)/erfa-$(version).tar.gz", "80239ae9ba4bf7f328f7f64baec7fc38e587058b3ef9cea9331c751de0b98783")
```

**Comment by giordano** [comment]
> Odd failures 🤔 

I believe we currently have a misconfiguration which causes all build jobs to download again the sources, in parallel, causing race conditions, and in particular the downloaded file to be overwritten all the time. Hopefully this will be fixed next week

**Pull request merged**